scripts/_common.sh

```diff
@@ -6,7 +6,7 @@
 pkg_dependencies="php5-gd php5-json php5-intl php5-mcrypt php5-curl php5-apcu php5-redis php5-ldap php5-imagick imagemagick acl tar smbclient"
 
 if [ "$(lsb_release --codename --short)" != "jessie" ]; then
-	pkg_dependencies="$pkg_dependencies php-zip"
+	pkg_dependencies="$pkg_dependencies php-zip php-apcu php-mbstring php-xml"
```

Comment:
I have found a idea to fix the issue of "how could we define that we need only this package on jessie" : 

Just write by example (for php-zip only on jessie) : `"php-zip | base-files>=9.0"`
"base-file" is : "Debian base system miscellaneous files"

I know that it's really late but I prefer to share this idea because I think it could solve many actual problem.

Reply:
Actually, I rather prefer a ` if not jessie then do that` instead of the syntax you suggest. Because, I think that is clearer to have the current syntax. And you know that I like to keep the code clear and readable ;)

Reply:
Yes I really understand your point of view but with that we don't need to upgrade the app after the migration to stretch. By that we might probably avoid some migration problem about from the dependence. It's from my point of view the main advantage of that.

But it's stay a suggestion.

Reply:
Do you mean that a simple apt upgrade will install the missing packages during the migration to stretch ?

Reply:
Yes probably. It's the reason because I purposed that.

Reply:
"probably" ? ;)
Could be nice to be sure about that.

Reply:
I say probably because I did't try the migration but I have tried [in friendica package](https://github.com/YunoHost-Apps/friendica_ynh/commit/2706331022719e7ed355baa85fd38a4ac9ba5e6c) the install in jessie and stretch and it was good.

About the migration yes I haven't tried yet but I really don't see any reason that I won't work because with this concept we Install always the same package and it's apt which will analyse the dependence.

Reply:
Anyway If you are really interested by that I could do a migration test but I don't want to lose my time for an unused things.

Reply:
We know that this issue could be a real problem during the migration, because in fact, it needs to upgrade each apps to install the dependencies.
So, if your solution works, it could be useful.

Reply:
Ok, is it ok for you if I do a migration test tomorrow with friendica which implement already that concept to validate completely that idea ?

Reply:
Yes.
However, would be even better to try with nextcloud, because we know that currently it doesn't work in case of migration without doing an upgrade.
But, nevertheless, the main thing is to be sure that the missing packages will be installed during the migration.

Reply:
I tried a migration and the problem that I had now was that I didn't updated all "php5" dependence. I think to solve completely the problem we need to use something like that for each "php5" dependences : 
`php5-curl|php-curl` to say to apt to use the php7 version in stretch for each dependence.
I could try a new time a migration I your stay interested on that.

With something like that it should work : 
```
pkg_dependencies="php5-gd|php-gd php5-json|php-json php5-intl|php-intl php5-mcrypt|php-mcrypt php5-curl|php-curl php5-apcu|php-apcu php5-redis|php-redis php5-ldap|php-ldap php5-imagick|php-imagick imagemagick acl tar smbclient php-zip|base-files(<<9.0) php-apcu|base-files(<<9.0) php-mbstring|base-files(<<9.0) php-xml|base-files(<<9.0)"
```

Reply:
Why not since nextcloud is official.
But, we shouldn't expect anyone else to do the same, this syntax is far more too complex for a simple list of dependencies...
If it works though, a comment would much appreciate.

Reply:
So, I did a new test and know it work better. The last problem is that apt is lazy so it won't install by example `php7.0-gd` as long as `php5-gd` is installed. But for the others it's ok.

Reply:
So when migrating, php5-* packages are kept and this package just can't work with dependencies handled like this, can it?

Reply:
I tested with roundcube and nextcloud and after the migration both app worked again without any upgrade but I think we cant guarantee that all package are installed. The solution that I found to force apt to install all new dependence is to hold the package (nextcloud_ynh_deps) and remove the php5-* package and in this case apt will automatically install the new dependence.

Reply:
Not sure that would be better than an simple upgrade of the app in this case...

Reply:
> Not sure that would be better than an simple upgrade of the app in this case...

Me too

Reply:
Considering that it's only a problem with the migration to stretch, I think really that we can keep our current if/else. We're not going to have a migration everyday. So let's keep it clear and easy.